.github/workflows/jest-coverage-changes.yml

```diff
@@ -0,0 +1,27 @@
+name: Jest Coverage - changed files
+
+on:
+  pull_request:
+    branches: develop
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+      - run: |
+          git fetch origin develop
+          git checkout -b develop origin/develop
+          git checkout ${{ github.event.pull_request.head.sha }}
+
+      - uses: actions/setup-node@v4
+        with:
+          node-version: lts/*
+      - name: Install dependencies
+        run: cd frontend && npm install -g yarn && yarn
+
+      - name: npm run test:changedsince
+        run: cd frontend && npm run i18n:generate-hash && npm run test:changedsince
```

Comment:
changedSince can accept a branch and we can specify `origin/develop` rather than checking out the code